### PR TITLE
Unsaved indicator for messages

### PR DESCRIPTION
**Pull request by elayabharath**
- editing message
- editing group, category

### CODE REVIEW

**Review by marianamarasoiu — CHANGES_REQUESTED** [review]
thanks EB - same comments as for https://github.com/larksystems/nook/pull/643 - looking forwards to reviewing it again!

File: webapp/web/configure/messages.css [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+.standard-message.unsaved .message::before {
+  font-family: "Font Awesome 5 Free";
+  content: "\f071";
+  display: inline-block;
+  padding-right: 3px;
+  vertical-align: middle;
+  color: var(--orange);
+  font-weight: 900;
+  margin-right: 3px;
+}
```
Comment (marianamarasoiu) [inline webapp/web/configure/messages.css:15]: nit - GH was complaining about this:
```suggestion
}

```

File: webapp/lib/app/configurator/standard_messages/controller.dart [hunk 2]
```diff
@@ -117,41 +119,51 @@ class MessagesConfiguratorController extends ConfiguratorController {
...
         _addMessagesToView({groupData.category: {newGroup.groupDescription: []}}, startEditingName: true);
+        _updateUnsavedIndicators(standardMessagesManager.categories, standardMessagesManager.unsavedMessageIds, standardMessagesManager.unsavedGroupIds, standardMessagesManager.unsavedCategoryIds);
         break;
 
       case MessagesConfigAction.updateStandardMessagesGroup:
         StandardMessagesGroupData groupData = data;
         standardMessagesManager.renameStandardMessageGroup(groupData.category, groupData.group, groupData.newGroupName);
         _view.categoriesByName[groupData.category].renameGroup(groupData.group, groupData.newGroupName);
+        _view.categoriesByName[groupData.category].markAsUnsaved(true);
+        _view.categoriesByName[groupData.category].groupsByName[groupData.newGroupName].markAsUnsaved(true);
```
Comment (marianamarasoiu) [inline webapp/lib/app/configurator/standard_messages/controller.dart:137]: do we still need these `markAsUnsaved(true)` if we call `_updateUnsavedIndicators` right after?

File: webapp/lib/app/configurator/standard_messages/controller.dart [hunk 3]
```diff
@@ -117,41 +119,51 @@ class MessagesConfiguratorController extends ConfiguratorController {
...
+        _updateUnsavedIndicators(standardMessagesManager.categories, standardMessagesManager.unsavedMessageIds, standardMessagesManager.unsavedGroupIds, standardMessagesManager.unsavedCategoryIds);
         break;
 
       case MessagesConfigAction.removeStandardMessagesGroup:
         StandardMessagesGroupData groupData = data;
         standardMessagesManager.deleteStandardMessagesGroup(groupData.category, groupData.group);
         _view.categoriesByName[groupData.category].removeGroup(groupData.group);
+        _view.categoriesByName[groupData.category].markAsUnsaved(true);
```
Comment (marianamarasoiu) [inline webapp/lib/app/configurator/standard_messages/controller.dart:145]: same question as above

**Review by marianamarasoiu — APPROVED** [review]
thanks EB - LGTM! just a couple of questions and a small nit

File: webapp/lib/app/configurator/standard_messages/controller.dart (hunk 3 above)
Comment (elayabharath) [reply webapp/lib/app/configurator/standard_messages/controller.dart]: My bad, there are a couple places more where they are not necessary.

**Pull request merged**